test/ranges.jl

```diff
@@ -2112,3 +2112,16 @@ end
 @test length(range(1, 100, length=big(100)^100)) == big(100)^100
 @test length(range(big(1), big(100)^100, length=big(100)^100)) == big(100)^100
 @test length(0 * (1:big(100)^100)) == big(100)^100
+
+@testset "issue #41784" begin
+    # test `in` when step equals 0
+    # test for Int
+    x = abs(rand(Int))
+    @test (x in StepRangeLen(x, 0, abs(rand(Int)))) == true
+    @test ((x - 1) in StepRangeLen(x, 0, abs(rand(Int)))) == false
+
+    # test for Char
+    x = rand(Char)
+    @test (x in StepRangeLen(x, 0, abs(rand(Int)))) == true # true case
+    @test ((x - 1) in StepRangeLen(x, 0, abs(rand(Int)))) == false # false case
```

Comment:
AFAICT, `x` has a non-zero probability of being set to `'\0'` here for which `x-1` fails. Just pick an arbitrary, fixed `x`?
(And the remarks for the `Int` test case apply similarly, too, of course.)

Reply:
Ah, that's right. I set it to `x - 1` because I accounted for the probability of it overflowing, but ended up missing this. I'll just set it to some arbitrary value like you suggested instead!